test/test_client.py

```diff
@@ -0,0 +1,158 @@
+import json
+
+import pytest
+import app
+import requests
+from dotenv import load_dotenv, find_dotenv
+
+
+@pytest.fixture
+def client():
+    # Use our test integration config instead of the 'real' version
+    file_path = find_dotenv('.env.test')
+    load_dotenv(file_path, override=True)
+
+    # Create the new app.
+    test_app = app.create_app()
+    #
+    #     # Use the app to create a test_client that can be used in our tests.
+    with test_app.test_client() as client:
+        yield client
+
+
+class MockResponse(object):
+    def __init__(self, status_code, body_content):
+        self._status_code = status_code
+        self._body_content = body_content
+
+    def json(self):
+        cards_with_json = json.loads(json.dumps(self._body_content))
+        cards = []
+        for card_json in cards_with_json:
+            cards.append(json.loads(card_json))
+        return cards
+
+    @property
+    def status_code(self):
+        return self._status_code
+
+
+def test_index_page(monkeypatch, client):
+    def fake_get(*args, **kwargs):
+        cards = [generate_one_card_json('apples')]
+        return MockResponse(200, cards)
+
+    monkeypatch.setattr(requests, 'get', fake_get)
+    response = client.get('/')
+    response_html = response.data
+
+    assert 'apples' in response_html.decode("utf-8")
+    assert 'TODO' in response_html.decode("utf-8")
+
+
+def test_add_item_when_item_already_exists_should_redirect(monkeypatch, client):
+    def fake_get(*args, **kwargs):
+        cards = [generate_one_card_json('apples')]
+        return MockResponse(200, cards)
+
+    title = 'apples'
+    monkeypatch.setattr(requests, 'get', fake_get)
+    response = client.post('/', data=dict(title=title))
+    response_html = response.data
+
+    assert f'A card with the title {title} already exists.' in response_html.decode("utf-8")
+
+
+def test_add_item_when_item_is_empty_should_redirect(monkeypatch, client):
```

Comment:
Test name is wrong - you are actually expecting an error response containing an error message.